options.go

```diff
@@ -117,6 +146,8 @@ func buildOpts(options ...Option) *opts {
 	for _, option := range options {
 		option.apply(opts)
 	}
+
+	opts.defaults()
```

Comment:
Rather than applying the options first and then setting the default value after checking if it wasn't set, it's easier if we just keep the way it was before (set it to the default value first in initialization in L139, and apply method on the provided options can just override that).

Reply:
Make sense. Updated.